src/Unpacker.php

```diff
@@ -200,4 +216,20 @@ public function updateLock(Result $result, IOInterface $io): void
         }
         $this->composer->setLocker($locker);
     }
+
+
+    private function addLinks(Link $link, array &$links, string $linkType): void
+    {
+        $constraint = $link->getPrettyConstraint();
+        $constraint = substr($this->resolver->parseVersion($link->getTarget(), $constraint, false), 1) ?: $constraint;
+
+        $linkName = $link->getTarget();
+        $constraint = $this->versionParser->parseConstraints($constraint);
+
+        $links[$linkName] = [
```

Comment:
This looks wrong to me. You could have a `conflict` and a `require-dev` on the same package. Using the same `$links` indexed by package name won't work fine.